src/read.cpp

```diff
@@ -38,39 +49,40 @@ CharacterVector read_lines_(List sourceSpec, List locale_, std::vector<std::stri
   LocaleInfo locale(locale_);
   Progress progressBar;
 
-  R_len_t n = (n_max < 0) ? 10000 : n_max;
-  CharacterVector out(n);
+  bool chunked = chunk_size != -1;
 
-  R_len_t i = 0;
-  for (Token t = tokenizer.nextToken(); t.type() != TOKEN_EOF; t = tokenizer.nextToken()) {
-    if (progress && (i + 1) % 25000 == 0)
-      progressBar.show(tokenizer.progress());
+  std::vector<SEXP> out;
```

Comment:
This seems really dangerous to me - who is responsible for protecting the SEXPs against gc?